### PR TITLE
Bump the required Go version

### PR DESCRIPTION
**Pull request by iamqizhao**

### CODE REVIEW

**Comment by broady** [comment]
Could you add a sentence saying that Go 1.5+ is required and only the last two major versions of Go are supported?

**Comment by dsymonds** [comment]
I don't think we need to state the "last two major versions" thing. That doesn't need to be a policy here, and this code won't necessarily break just because Go 1.7 comes out. We should just bump the supported version here when this package wants to use something from a newer version, or a dependency forces us to.

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ $ go get google.golang.org/grpc
 Prerequisites
 -------------
 
-This requires Go 1.4 or above.
+This requires Go 1.5+.
```
Comment (dsymonds) [inline README.md]: I'd say "This requires Go 1.5 or later."; it's a little clearer than the "+".

Comment (iamqizhao) [inline README.md]: done.

**Comment by iamqizhao** [comment]
I signed it!

**Pull request merged**